### PR TITLE
Merge gz-fuel-tools8 ➡️  main

### PR DESCRIPTION
**Pull request by scpeters**
# ➡️  Forward port

Port `gz-fuel-tools8 ` ➡️  `main`

Branch comparison: https://github.com/gazebosim/gz-fuel-tools/compare/main...gz-fuel-tools8

**Note to maintainers**: Remember to **Merge** with commit (not squash-merge or rebase)

### CODE REVIEW

File: include/gz/fuel_tools/config.hh.in [hunk 1]
```diff
@@ -30,6 +30,6 @@
 
 #define GZ_FUEL_TOOLS_VERSION_HEADER "Gazebo Fuel Tools, version ${PROJECT_VERSION_FULL}\nCopyright (C) 2017 Open Source Robotics Foundation.\nReleased under the Apache 2.0 License.\n\n"
 
-#define GZ_FUEL_INITIAL_CONFIG_PATH "${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_DATAROOTDIR}/ignition/${GZ_DESIGNATION}${PROJECT_VERSION_MAJOR}/"
+#define GZ_FUEL_INITIAL_CONFIG_PATH _Pragma ("GCC warning \"'GZ_FUEL_INITIAL_CONFIG_PATH' macro is deprecated and should not be used. \"") "${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_DATAROOTDIR}/ignition/${GZ_DESIGNATION}${PROJECT_VERSION_MAJOR}/"
```
Comment (azeey) [inline include/gz/fuel_tools/config.hh.in:33]: I think this can be removed.

File: src/cmd/CMakeLists.txt [hunk 2]
```diff
@@ -1,17 +1,60 @@
...
-if (APPLE)
-  set(GZ_LIBRARY_NAME lib${PROJECT_NAME_LOWER}.dylib)
-elseif(MSVC)
-  set(GZ_LIBRARY_NAME ${PROJECT_NAME_LOWER}.dll)
-else()
-  set(GZ_LIBRARY_NAME lib${PROJECT_NAME_LOWER}.so)
-endif()
+# Unlike other gz libraries, the ruby script for the fuel_tools library is called cmdfuel.rb instead of cmdfuel_tools.rb
+set(CMD_NAME cmdfuel)
+set(cmd_script_generated_test "${CMAKE_BINARY_DIR}/test/lib/$<CONFIG>/ruby/ignition/${CMD_NAME}${PROJECT_VERSION_MAJOR}.rb")
```
Comment (azeey) [inline src/cmd/CMakeLists.txt:7]: ignition -> gz?

**Comment by scpeters** [comment]
`FuelClientTest/FuelClientDownloadTest.DownloadModel/1` is failing in multiple CI platforms:

* https://build.osrfoundation.org/job/ignition_fuel-tools-ci-pr_any-ubuntu_auto-amd64/1057/testReport/junit/(root)/FuelClientTest_FuelClientDownloadTest/DownloadModel_1/
* https://build.osrfoundation.org/job/ignition_fuel-tools-ci-pr_any-homebrew-amd64/1036/testReport/junit/(root)/FuelClientTest_FuelClientDownloadTest/DownloadModel_1/

File: src/cmd/CMakeLists.txt (hunk 2 above)
Comment (scpeters) [reply src/cmd/CMakeLists.txt]: I amended the merge commit to https://github.com/gazebosim/gz-fuel-tools/pull/363/commits/31bb99766fce404b14fdf1467e5bd1f79950e3c5 with this change

**Comment by scpeters** [comment]
as in https://github.com/gazebosim/gz-fuel-tools/pull/362#issuecomment-1644418337, the windows test failures are already present on the `main` branch

* https://build.osrfoundation.org/job/ign_fuel-tools-ci-win/53/testReport/

**Pull request merged**